src/mpyl/steps/deploy/k8s/kubectl.py

```diff
@@ -0,0 +1,85 @@
+""" This module is called on to push your kubernetes manifest to a(n) (argo) repo during the `mpyl.steps.deploy step."""
+
+import shutil
+from logging import Logger
+from pathlib import Path
+from tempfile import TemporaryDirectory
+
+from .rancher import ClusterConfig
+from ....project import Target
+from ....utilities.repo import RepoConfig, Repository
+from ....steps import Input, Output
+from ....utilities.subprocess import custom_check_output
+
+
+def push_manifest_to_repo(
+    logger: Logger,
+    step_input: Input,
+    rancher_config: ClusterConfig,
+    manifest_path: Path,
+) -> Output:
+    git_config = step_input.run_properties.config["vcs"]["argoRepository"]
+    if not git_config:
+        raise ValueError("No argocd repository configured")
+    argocd_repo_config = RepoConfig.from_git_config(git_config=git_config)
+    with TemporaryDirectory() as tmp_repo_dir:
+        with Repository.from_clone(
+            config=argocd_repo_config, repo_path=Path(tmp_repo_dir)
+        ) as artifact_repo:
```

Comment:
By the way, we use the name artifact_repo for both the caching and the argo ones. I would name the variables more consistently